pylint/pyreverse/diagrams.py

```diff
@@ -225,9 +225,13 @@ def extract_relationships(self) -> None:
                         value, obj, name, "aggregation"
                     )
 
-            for name, values in list(node.associations_type.items()) + list(
-                node.locals_type.items()
-            ):
+            associations = node.associations_type.copy()
+
+            for name, values in node.locals_type.items():
+                if name not in associations:
+                    associations[name] = values
+
+            for name, values in associations.items():
```

Comment:
Couldn't we use ``set(list(node.associations_type.items()) + list(node.locals_type.items()))``?

Reply:
No, that gives `TypeError: unhashable type: 'list'`. The types involved here are still a little obscure to me. I tried a few other variations and this seems to be the simplest one that works.